### ISSUE TITLE
[IDEA] "size=max" to expand edittext widget horizontally

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#7567]
**Is your feature request related to a problem? Please describe.**
Two *very* common scenarios for the **edittext** widget:
1) fully expand its width to fill the container space horizontally
2) same as (1) but there's also a *button* next to it (to "set" som value that is entered into the edittext widget) and the editor+button should stay in the same unbroken row

Currently, using the widgets `size` attribute results in a static and non-responsive width. And there is no direct inline `style` attribute so the user must use the `class` attribute and define it elsewhere. Using `width:100%` solves case (1) but, for case (2), `width:100%` pushes the button to the next row, and a lot more complex trickery is required to get around this. 

**Describe the solution you'd like**
The native attribute to set the widgets width is `size`:

>size: The size of the input field (in characters). This exact result depends on browser and font. Use the class attribute to style width for precise control

I propose an attribute value`size=max` that expands the field to satisfy problems (1) and (2)

**Comment by pmario** [comment]
For 1) you can use `<$edit-text tiddler=test tag=input class="tc-max-width"/>`

2) would be easy to do with a table and  [table-utility classes](https://tiddlywiki.com/#Tables%20in%20WikiText%20CSS%20Utility%20Classes) which we did develop to gether. 

```
|tc-max-width  tc-edit-max-width tc-table-no-border tc-first-link-nowrap|k|k
| <$button class="xx">button with more text</$button>|<$edit-text tiddler=test tag=input class="tc-max-width"/>|
| <$button class="xx">asdf</$button>|<$edit-text tiddler=test tag=input class="tc-max-width"/>|
```
I'm not 100% sure, what exactly you want to achieve but imo tables work well here, without a lot of new CSS rules.

**Comment by pmario** [comment]
I did some experiments with buttons. .. We probably do need some more utility classes. ... not sure yet, if there is an easier way

**Comment by twMat** [comment]
@pmario 

>1) `class="tc-max-width`

Fair enough *if* it is mentioned in the doc tid for the widget, the class attribute description. I'll add a PR.

>2) would be easy to do with a table and table-utility classes

Not even close to "fair enough". Even if the user comes up with that solution - which no normal user would - then it is still *way* to convoluted involving tables and stuff. Again, the scenario I describe is very common and needs a (truly) simple solution. IMO `size=max` or something along those lines, is reasonable.

**Comment by pmario** [comment]
> IMO size=max or something along those lines, is reasonable.

If you know how to achieve it with CSS .. Let me know.

**Comment by pmario** [comment]
The best and consistent looking **responsive form-linke** configuration that is easy to create with wikitext is this. STYLE is only there for easy testing. In production it would be a Stylesheet tiddler.

It's possible to easily arrange the buttons left or right. 
If there is an input area the button can be moved to the top or bottom .. whatever. 

```
<style>
.tc-col-fit-content td:nth-child(1) {
  width: min-content;
}
</style>

|tc-max-width tc-col-fit-content tc-max-width tc-edit-max-width |k
|<$button class="xx">button with more text</$button> |<$edit-text tiddler=test tag=input class="tc-max-width"/>|
| <$button class="xx">asdf</$button>|<$edit-text tiddler=test tag=input class="tc-max-width"/>|
|^ <$button class="xx">Do Something</$button>|<$edit-text tiddler=test class="tc-max-width"/>|
```

![image](https://github.com/Jermolene/TiddlyWiki5/assets/374655/48c3b9e5-fed3-4d00-9a5a-09850b7b57db)

```
<style>
.tc-col-fit-content td:nth-child(1) {
  width: min-content;
}
</style>

|tc-max-width tc-col-fit-content tc-max-width tc-edit-max-width |k
|<$button class="xx">button with more text</$button> |Some text above the input area will give the right sie more space. <$edit-text tiddler=test tag=input class="tc-max-width"/>|
| <$button class="xx">asdf</$button>|<$edit-text tiddler=test tag=input class="tc-max-width"/>|
|^ <$button class="xx">Do Something</$button>|<$edit-text tiddler=test class="tc-max-width"/>|
```

looks like this. I think it **looks good**, is **responsive** and can easily be **achieved with existing wikitext**. 
The only thing missing is some more documentation, how to achieve it. 

![image](https://github.com/Jermolene/TiddlyWiki5/assets/374655/08abbb7d-68f9-4021-864b-c15596346ccc)

**Comment by pmario** [comment]
For my script-manager, which I use every day to collect links and text-snippes I did create a grid-based input area, that also looks like a table. 

It's more adjustable, but also much more specifically designed for exactly that usecase. 

![image](https://github.com/Jermolene/TiddlyWiki5/assets/374655/8b28673b-3e7e-424a-8642-aae1bbd4736c)

**Comment by twMat** [comment]
Yeah... unfortunately, it just expects so much from the user.

Might it be possible to define a "container" class, e.g `tc-single-line`, styled with `display:inline-table` or something  (`display` has many [possible values](https://www.w3schools.com/cssref/pr_class_display.php)) and so any inner elements would somehow be targeted and get `display:table-cell`, along with the other styling for positioning?

Then, if that is at all possible, the user could do something like

```
@@.tc-single-line
<$edit-text .... />
<$button ... />
@@
```

(Or - still assuming the above works at all - if we modernize our CSS approach, we could define custom html tags, such as `<tw-single-line>` which has the advantage of being more semantic than the div and it appears intentionally designed in TW as a "feature" rather than just as a "style".)

**Comment by twMat** [comment]
(I should have added a call to: @pmario , so I'm adding it here instead)

**Comment by twMat** [comment]
@pmario here's a better version

```
@@.tc-expand-input
<$edit-text tiddler=test tag=input />
<$button>button with more text</$button>
@@

<style>
.tc-expand-input {display:inline-flex; width:100%;}
.tc-expand-input input {flex-grow:1;}
</style>
```

**Comment by twMat** [comment]
@pmario - I think this might be a solution. It uses the existing `tc-edit-max-width`. 

```
@@.tc-edit-max-width 
<$edit-text tiddler=test tag=input />
<$button>button with more text</$button>
@@

<style>
.tc-edit-max-width:has(input) {display:flex;}
.tc-edit-max-width:has(input) input {height:min-content;}
</style>
```

The `height:min-content` is needed because otherwise the other element (the button) controls the height of the editor - on my screen the button label wraps unless I also add `.tc-edit-max-width :not(input) {white-space:nowrap}`, but so instead I set the height for the editor.

...or maybe the checking of `:has(input)` is not needed - this also works:

```
<style>
.tc-edit-max-width {display:flex;}
.tc-edit-max-width input {height:min-content;}
```

Could `tc-edit-max-width` be extended to feature this?

BTW, the only definition I find in TW for `tc-edit-max-width` is this:

```
/* Allow input and textarea to look like the ControlPanel inputs */
.tc-edit-max-width input,
.tc-edit-max-width textarea {
	width: 100%;
	padding: 3px;
}
```
